components/CompanyDetailsCard/CompanyDetailsCard.tsx

```diff
@@ -19,17 +24,89 @@ type CompanyDetailsCardProps = {
   company: Company;
 };
 
+function extractDomain(url: string) {
+  const parts = url?.split("www.");
+  if (parts?.length >= 2) {
+    const domain = parts[1];
+    return domain.replace(/\/$/, ""); // Remove trailing "/"
+  } else {
+    return null; // "www." not found in the URL
+  }
+}
+
 const CompanyDetailsCard: React.FC<CompanyDetailsCardProps> = ({ company }) => {
+  const [displayWebsite, setDisplayWebsite] = useState(false);
   return (
     <div className={styles.detailsCardMainContainer}>
-      <h2>{company?.name}</h2>
-      <h4>City: {company?.city}</h4> <h4>Activity: {company?.category}</h4>
-      <h4> {company?.website}</h4>
-      <h5>IND number: {company?.indReferentNumber}</h5>
-      <h5>
-        Address:{" "}
-        {`${company?.street}, ${company?.houseNumber}. ${company?.city}. ${company?.postCode}.`}
-      </h5>
+      <div className={styles.topBar}>
+        <Logo />
+        <div className={styles.avatar}>
+          <Avatar imageSource={userImage} className="detailsPage" />
+        </div>
+      </div>
+
+      <div className={styles.companyName}>
+        <Link href="/">
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="28"
+            height="28"
+            viewBox="0 0 28 28"
+            fill="none"
+          >
+            <g opacity="0.5">
+              <path
+                d="M17.5 21L10.5 14L17.5 7"
+                stroke="#1E213F"
+                stroke-width="2"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+              />
+            </g>
+          </svg>
+        </Link>
+
+        <Heading1 variant="detailsPage">{company?.name}</Heading1>
+      </div>
+      <div className={styles.infos}>
+        <Heading1 variant="detailsLabel">Location</Heading1>
```

Comment:
You are using `<h1>` elements here for all details. But there should be a single `h1` for a page. I some cases you multiple `h1` on one page, when you have a series of articles or blogposts, each one can have a `h1`. But here all the `h1` are not really a good idea. 

This type of element is more like a label (for Location, Area, Website and so on), so maybe create a reusable `label` for that.  
And then the actual value is more like a `<p>` so maybe create a reusable paragraph for that